### ISSUE TITLE
Decide what to do with variant marshalling

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtime#72911]
We have a couple libs tests (such as `GetObjectForNativeVariant_ThrowsNotSupportedException`) that expect variant marshalling to throw when COM is disabled. We have partial support for variant marshalling. Either delete it fully from NativeAOT or adapt the tests such that they only fail if COM is actually needed (but this wouldn't match the CoreCLR behavior).

**Comment by jkotas** [comment]
The variant marshalling was done by @kant2002 in https://github.com/dotnet/runtimelab/pull/1182 to make more of WinForms work.

As we work through making WinForms work trimming and built-in COM disabled, we will see what's the best uniform course of the action here - for both CoreCLR w/ trimming and NativeAOT.

**Comment by MichalStrehovsky** [comment]
Yup, it's likely the resolution for this will fall out from the WinForms work since we'll want WinForms to work with COM disabled on CoreCLR. I just needed some issue to block the tests on that is in the Future milestone and we don't plan to do anything for in a while.

**Comment by kant2002** [comment]
Based on my experience, I would like to ask for proper Variant implementation. Right now all of them private and it seems half done (in a sense obviously), both in NativeAOT and in WinForms. There a lot of edge cases from original interop, and maintaining them is painful. I like NativeAOT implementation better, even if WinForms Variant was slightly easier to understand.
